src/coreclr/jit/compiler.h

```diff
@@ -4615,7 +4615,10 @@ class Compiler
     PhaseStatus placeLoopAlignInstructions();
 #endif
 
-    GenTree* fgInitThisClass();
+    // We use this field to set the desired R2R helper call, which can be the constructor
+    // helper, or the same helper we are using the code, so the CSE will eliminate one of both.
+    CorInfoHelpFunc m_preferredInitCctor;
```

Comment:
Nothing is initializing this from what I can see, maybe that's the cause of your issues?

I think you can initialize it from `compCompile`, there is already one place there that checks for `CORINFO_NATIEVAOT_ABI`.